openwisp_controller/geo/api/views.py

```diff
@@ -186,6 +187,61 @@ class GeoJsonLocationList(
     filterset_class = LocationOrganizationFilter
 
 
+class IndoorCoodinatesViewPagination(ListViewPagination):
+    page_size = 50
+
+
+class IndoorCoordinatesList(ProtectedAPIMixin, generics.ListAPIView):
+    serializer_class = IndoorCoordinatesSerializer
+    filter_backends = [filters.DjangoFilterBackend]
+    pagination_class = IndoorCoodinatesViewPagination
+    queryset = (
+        DeviceLocation.objects.filter(
+            location__type="indoor",
+            floorplan__isnull=False,
+        )
+        .select_related(
+            "content_object", "location", "floorplan", "content_object__organization"
+        )
+        .order_by("floorplan__floor")
+    )
+
+    def get_available_floors(self):
+        location_id = self.kwargs.get("pk")
+        floors = list(
+            self.queryset.filter(location__id=location_id)
+            .values_list("floorplan__floor", flat=True)
+            .distinct()
+        )
+        return floors
+
+    def get_floor_from_query_or_default(self, qs, floor_param):
```

Comment:
Sorry I didn't spot it earlier. We should implement the filtering using the django_filters backend. 

You can look here for reference 
https://github.com/openwisp/openwisp-controller/blob/4ea68be945a3b7fc15146b9d47a79f315d5f682d/openwisp_controller/geo/api/views.py#L43-L51

It is also possible to customize the filtering. You can look at `TemplateListFilter` and refer django-filters docs.